applications/serial_lte_modem/src/slm_at_commands.c

```diff
@@ -0,0 +1,580 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr/types.h>
+#include <ctype.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <logging/log.h>
+#include <drivers/uart.h>
+#include <string.h>
+#include <init.h>
+#include <modem/at_cmd.h>
+#include <modem/at_cmd_parser.h>
+#include <modem/at_notif.h>
+#include <power/reboot.h>
+
+#include "slm_util.h"
+#include "slm_at_host.h"
+#include "slm_at_tcp_proxy.h"
+#include "slm_at_udp_proxy.h"
+#include "slm_at_tcpip.h"
+#if defined(CONFIG_SLM_NATIVE_TLS)
+#include "slm_at_cmng.h"
+#endif
+#include "slm_at_icmp.h"
+#include "slm_at_fota.h"
+#if defined(CONFIG_SLM_GPS)
+#include "slm_at_gps.h"
+#endif
+#if defined(CONFIG_SLM_FTPC)
+#include "slm_at_ftp.h"
+#endif
+#if defined(CONFIG_SLM_MQTTC)
+#include "slm_at_mqtt.h"
+#endif
+#if defined(CONFIG_SLM_HTTPC)
+#include "slm_at_httpc.h"
+#endif
+
+LOG_MODULE_REGISTER(slm_at, CONFIG_SLM_LOG_LEVEL);
+
+/**@brief Shutdown modes. */
+enum shutdown_modes {
+	SHUTDOWN_MODE_IDLE,
+	SHUTDOWN_MODE_SLEEP,
+	SHUTDOWN_MODE_INVALID
+};
+
+/**@brief AT command handler type. */
+typedef int (*slm_at_handler_t) (enum at_cmd_type);
+
+/**@brief AT command list item type. */
+typedef struct slm_at_cmd {
+	char *string;
+	slm_at_handler_t handler;
+} slm_at_cmd_t;
+
+static struct slm_work_t {
```

Comment:
No need for `_t` suffix, you don't use typedef here.

Reply:
updated